### PR TITLE
Improved method checks in enum declaration

### LINKED ISSUE #5086: [PHP] Not relevant enum methods must produce error
### Apache NetBeans version

Apache NetBeans 16

### What happened

Not relevant enum methods must produce error

### How to reproduce

```php
enum Suit: string {
  case Hearts = 'H';
  case Diamonds = 'D';
  case Clubs = 'C';
  case Spades = 'S';

  // already correct error message display
  public function __construct($param) {
    echo "__construct";
  }

  // this statement must display error
  public function __destruct() {
    echo "__destruct";
  }

  // this statement must display error
  public function __set($arg1, $arg2) {
    echo "__set";
  }

  // this statement must display error
  public function __get($arg1) {
    echo "__get";
  }

  // OK
  public function __call(string $name, array $args) : mixed {
    echo "__call";
  }

  // OK
  public static function __callStatic(string $name, array $arguments): mixed {
    echo "__callStatic";
  }
}
```

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 10

### JDK

Java: 14.0.1; Java HotSpot(TM) 64-Bit Server VM 14.0.1+7 Runtime: Java(TM) SE Runtime Environment 14.0.1+7

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

_No response_

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

### LINKED ISSUE #5094: [PHP] Manually defining a Enum::cases() is forbidden
### Apache NetBeans version

Apache NetBeans 16

### What happened

Manually defining a Enum::cases() is forbidden.

### How to reproduce

```php
<?php
enum Suit: string {
  case Hearts = 'H';
  case Diamonds = 'D';
  case Clubs = 'C';
  case Spades = 'S';
  public function case() {}  // this statement must display error
  public function other() {}  // OK
}


### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 10

### JDK

Java: 14.0.1; Java HotSpot(TM) 64-Bit Server VM 14.0.1+7 Runtime: Java(TM) SE Runtime Environment 14.0.1+7

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

_No response_

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

### PR DESCRIPTION
**Pull request by KacerCZ**
- Checks declaration of forbidden methods.
- Adds __invoke and __debugInfo to list of magic methods.

Docs:
- https://www.php.net/manual/en/language.enumerations.listing.php
- https://www.php.net/manual/en/class.backedenum.php
- https://www.php.net/manual/en/language.enumerations.object-differences.php
- https://www.php.net/manual/en/language.oop5.magic.php

Closes  #5086, closes  #5094

### CODE REVIEW

**Comment by mbien** [comment]
fyi: if you want to get bug fixes into NB17 there is still a chance during the RC phase. All you have to do is to rebase on `delivery` and change the PR target to `delivery` too.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java [hunk 1]
```diff
@@ -142,13 +167,22 @@ private void checkTraits(Collection<? extends TraitScope> traits, EnumScope enum
...
         for (MethodScope method : methods) {
             if (CancelSupport.getDefault().isCancelled()) {
                 return;
             }
             if (method.isConstructor()) {
                 addHint(method, Bundle.IncorrectEnumHintError_incorrectEnumConstructor(), hints);
+                continue;
+            }
+            String methodName = method.getName().toLowerCase();
+            if ("cases".equals(methodName)) {
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java:180]: `NOI18N`

**Comment by junichi11** [comment]
We should add the following URLs and issue numbers to the commit message.

> Docs:
> 
> * https://www.php.net/manual/en/language.enumerations.listing.php
> * https://www.php.net/manual/en/class.backedenum.php
> * https://www.php.net/manual/en/language.enumerations.object-differences.php
> * https://www.php.net/manual/en/language.oop5.magic.php

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java (hunk 1 above)
Comment (KacerCZ) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java]: Fixed.

**Comment by KacerCZ** [comment]
Thanks for the review.
I've updated the commit message.

**Comment by junichi11** [comment]
As @mbien wrote, I think we can change the target to NB17.

**Review by junichi11 — APPROVED** [review]

**Comment by neilcsmith-net** [comment]
@KacerCZ OK, please make sure this is based on top of delivery and update the PR base branch and milestone if you want this to go in to NB17.  Will keep an eye out for it.

**Pull request merged**